staging/src/k8s.io/api/authentication/v1/types.go

```diff
@@ -31,6 +31,9 @@ const (
 	// It can be repeated multiplied times for multiple groups.
 	ImpersonateGroupHeader = "Impersonate-Group"
 
+	// ImpersonateUIDHeader is used to impersonate a particular UID during an API server request
+	ImpersonateUIDHeader = "Impersonate-Uid"
```

Comment:
project standard capitalizes UID.  See `GetUID` as an example: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go#L501

Reply:
https://github.com/kubernetes/kubernetes/pull/99961#discussion_r605211372 I actually changed this earlier because it seems like the canonicalized form for headers is `Impersonate-Uid`. Should it be capitalized even in the header?

Reply:
[http.CanonicalHeaderKey](https://golang.org/pkg/net/http/#CanonicalHeaderKey) will force this to be `Impersonate-Uid` on incoming requests and we want to make sure that `request.Header["Impersonate-Uid"]` works (`request.Header.Get("Impersonate-Uid")` always works because `Get` redoes the canonicalization).  IMO this is correct as-is.

```go
type Request struct {
	// Header contains the request header fields either received
	// by the server or to be sent by the client.
	//
	// If a server received a request with header lines,
	//
	//	Host: example.com
	//	accept-encoding: gzip, deflate
	//	Accept-Language: en-us
	//	fOO: Bar
	//	foo: two
	//
	// then
	//
	//	Header = map[string][]string{
	//		"Accept-Encoding": {"gzip, deflate"},
	//		"Accept-Language": {"en-us"},
	//		"Foo": {"Bar", "two"},
	//	}
	//
	// ...
	//
	// HTTP defines that header names are case-insensitive. The
	// request parser implements this by using CanonicalHeaderKey,
	// making the first character and any characters following a
	// hyphen uppercase and the rest lowercase.
	//
	// ...
	Header Header

	// ...
}
```

Reply:
> we want to make sure that `request.Header["Impersonate-Uid"]` works

Eh, the only reason to do that is if you actually care about case, but ok